### PR TITLE
29 have a wallet encoding iso with massa client

### PR DESCRIPTION
**Pull request by SebastienValla**

### CODE REVIEW

File: pkg/wallet/wallet.go
Comment (gregLibert) [inline pkg/wallet/wallet.go:24]: Why? According to NIST, the salt must be at least 128 bits long. By reducing it to 96 bits, you reduce the number of attempts needed to test all values by 2^36, or 68,719,476,736.

Source to do the math: Why? According to NIST, the salt must be at least 128 bits long. By reducing it to 96 bits, you reduce the number of attempts needed to test all values by 2^36, or 68,719,476,736.

**Review by gregLibert — CHANGES_REQUESTED** [review]
:no_good: security reasons.

File: pkg/wallet/wallet.go
Comment (SebastienValla) [reply pkg/wallet/wallet.go:24]: so there is a safety issue in massa-client, I will contact the core team
https://github.com/massalabs/massa/blob/main/massa-cipher/src/constants.rs

Comment (AurelienFT) [reply pkg/wallet/wallet.go:24]: Our salt can't be more than 64 because it's the limit for the crate we use (we can change crate). Here is their explanations on why they placed 64 as a limit : https://docs.rs/password-hash/latest/password_hash/struct.Salt.html#constraints

Comment (gregLibert) [reply pkg/wallet/wallet.go:24]: Sounds good. We just need it to be 16 bytes long (128 bits). It's what we did on our side :-)

Comment (AurelienFT) [reply pkg/wallet/wallet.go:24]: Ok I misread "bits" and "bytes". We can change it to 16 on our side too

**Comment by SebastienValla** [comment]
waiting for core team to move on to unblock this issue

**Comment by SlnPons** [comment]
I think this PR will be unblocked once https://github.com/massalabs/massa-standards/issues/13 is done. 
@SebastienValla @Thykof so you agree?

**Comment by Thykof** [comment]
> I think this PR will be unblocked once [massalabs/massa-standards#13](https://github.com/massalabs/massa-standards/issues/13) is done. @SebastienValla @Thykof so you agree?

No, once https://github.com/massalabs/massa-standards/issues/15 is done

**Pull request closed without merging**
